tokio-util/src/io/stream_fd.rs

```diff
@@ -0,0 +1,144 @@
+use std::io;
+use std::io::{Error, Read, Write};
+use std::os::unix::io::AsRawFd;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio::io::unix::AsyncFd;
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+
+#[macro_use]
+mod ready {
+    macro_rules! ready {
```

Comment:
IMHO this should just be a public macro in `tokio-util` or even `tokio`.
It's so useful to manually writing `Future` but yet only `futures`/`futures-util` includes this.

Reply:
It's now in std but it doesn't meet our MSRV.

Reply:
I basically just copied in the private version within tokio.

Reply:
> It's now in std but it doesn't meet our MSRV.

I know and many crates still can't use them due to this, that's why I think tokio or tokio-util should have this.

Reply:
Good point. We should create a dedicated issue for this and figure out there which crate to put this in.

Do you wanna go ahead and file the issue?

Reply:
> Do you wanna go ahead and file the issue?

Thanks for the suggestion, I've opened a new issue for this.